torch/export/_trace.py

```diff
@@ -534,13 +534,54 @@ def _compiling_state_context():
     if isinstance(mod, torch.fx.GraphModule) and hasattr(mod, "meta"):
         gm.meta.update(mod.meta)
 
+
+    original_output_names = ()
+    for node in gm.graph.nodes:
+        if node.op == "output":
+            original_output_names = node.args
+
     if pre_dispatch:
         from torch._export.passes.replace_set_grad_with_hop_pass import (
             replace_set_grad_with_hop_pass,
         )
 
         gm = replace_set_grad_with_hop_pass(gm)
 
+
+    updated_output_names = ()
+    for node in gm.graph.nodes:
+        if node.op == "output":
+            updated_output_names = node.args
+
+    # It could be updated because replace_set_grad_with_hop_pass adds new
+    # getitem nodes into the graph.
```

Comment:
Right now it is only replace_set_grad_with_hop_pass, but in theory, it could be useful for any pass that adds submodules in the graph. I can make it a free function.